crates/ruff_python_parser/src/string.rs

```diff
@@ -278,16 +300,26 @@ impl<'a> StringParser<'a> {
     }
 
     fn parse_string(&mut self) -> Result<StringType, LexicalError> {
-        let mut value = String::new();
         let start_location = self.get_pos();
-        while let Some(ch) = self.next_char() {
-            match ch {
-                '\\' if !self.kind.is_raw() => {
-                    value.push_str(&self.parse_escaped_char()?);
-                }
-                ch => value.push(ch),
+        let mut value = String::new();
```

Comment:
Future (for another PR): It would be nice if we avoid allocating the string if input and output is the same.

Reply:
Agree, and provided input in the other comment.